server/src/main/java/org/eclipse/hono/server/HonoServer.java

```diff
@@ -254,16 +364,60 @@ public int getPort() {
     }
 
     /**
+     * Gets if Hono opens a secure port.
+     *
+     * @return The flag value for opening a secure port.
+     */
+    public boolean isOpenSecurePort() {
+        return openSecurePort;
+    }
+
+    /**
+     * Gets the insecure port Hono listens on for AMQP 1.0 connections (if configured).
+     * <p>
+     * If the port has been set to 0 Hono will bind to an arbitrary free port chosen by the operating system during
+     * startup. Once Hono is up and running this method returns the <em>actual port</em> Hono has bound to.
+     * </p>
+     *
+     * @return the insecure port Hono listens on.
+     */
+    public int getInsecurePort() {
+        if (!honoConfig.isAllowInsecure())
+            return HonoConfigProperties.PORT_UNCONFIGURED;
+        if (insecureServer != null) {
+            return insecureServer.actualPort();
+        } else {
+            return this.honoConfig.getInsecurePort();
+        }
+    }
+
+    /**
+     * Gets if Hono opens an insecure port.
+     *
+     * @return The flag value for opening an insecure port.
+     */
+    public boolean isOpenInsecurePort() {
+        return openInsecurePort;
+    }
+    /**
      * Gets the IP address Hono is bound to.
-     *  
+     *
      * @return The IP address.
      */
     public String getBindAddress() {
         return honoConfig.getBindAddress();
     }
 
+    /**
+     * Gets the IP address Hono's insecure port is bound to.
+     *
+     * @return The IP address.
+     */
+    public String getBindAddressInsecurePort() {
+        return honoConfig.getBindAddressInsecurePort();
+    }
+
     void handleRemoteConnectionOpen(final ProtonConnection connection) {
-        connection.setContainer(String.format("Hono-%s:%d", getBindAddress(), getPort()));
```

Comment:
Keep in mind, though, to use the correct bind address and port, depending on which port the peer connected to ...